server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -243,9 +243,23 @@ class DeltaSharingService(serverConfig: ServerConfig) {
   def getTableVersion(
     @Param("share") share: String,
     @Param("schema") schema: String,
-    @Param("table") table: String): HttpResponse = processRequest {
+    @Param("table") table: String,
+    @Param("startingTimestamp") @Nullable startingTimestamp: String
+  ): HttpResponse = processRequest {
     val tableConfig = sharedTableManager.getTable(share, schema, table)
-    val version = deltaSharedTableLoader.loadTable(tableConfig).tableVersion
+    if (startingTimestamp != null && !tableConfig.cdfEnabled) {
+      throw new DeltaSharingIllegalArgumentException("Reading table by version or timestamp is" +
+        " not supported because change data feed is not enabled on table: " +
+        s"$share.$schema.$table")
+    }
+    val version = deltaSharedTableLoader.loadTable(tableConfig).getTableVersion(
+      Option(startingTimestamp)
+    )
+    if (startingTimestamp != null && version < tableConfig.startVersion) {
+      throw new DeltaSharingIllegalArgumentException(
+        s"You can only query table data since version ${tableConfig.startVersion}."
```

Comment:
Perhaps also include the "version" that corresponds to the startTimestamp in the error message.

Reply:
updated.